otx/algorithms/segmentation/adapters/mmseg/data/dataset.py

```diff
@@ -0,0 +1,267 @@
+
+        self.pipeline = Compose(pipeline)
+
+    @staticmethod
+    @check_input_parameters_type()
+    def filter_labels(all_labels: List[LabelEntity], label_names: List[str]):
+        """Filtering Labels function."""
+        filtered_labels = []
+        for label_name in label_names:
+            matches = [label for label in all_labels if label.name == label_name]
+            if len(matches) == 0:
+                continue
+
+            assert len(matches) == 1
+
+            filtered_labels.append(matches[0])
+
+        return filtered_labels
+
+    def __len__(self):
+        """Total number of samples of data."""
+
+        return len(self.data_infos)
+
+    @check_input_parameters_type()
+    def pre_pipeline(self, results: Dict[str, Any]):
+        """Prepare results dict for pipeline."""
+
+        results["seg_fields"] = []
+
+    @check_input_parameters_type()
+    def prepare_train_img(self, idx: int) -> dict:
+        """Get training data and annotations after pipeline.
+
+        Args:
+            idx (int): Index of data.
+
+        Returns:
+            dict: Training data and annotation after pipeline with new keys introduced by pipeline.
+        """
+
+        item = self.data_infos[idx]
+
+        self.pre_pipeline(item)
+        out = self.pipeline(item)
+
+        return out
+
+    @check_input_parameters_type()
+    def prepare_test_img(self, idx: int) -> dict:
+        """Get testing data after pipeline.
+
+        Args:
+            idx (int): Index of data.
+
+        Returns:
+            dict: Testing data after pipeline with new keys introduced by pipeline.
+        """
+
+        item = self.data_infos[idx]
+
+        self.pre_pipeline(item)
+        out = self.pipeline(item)
+
+        return out
+
+    @check_input_parameters_type()
+    def get_ann_info(self, idx: int):
+        """This method is used for evaluation of predictions.
+
+        The CustomDataset class implements a method
+        CustomDataset.evaluate, which uses the class method get_ann_info to retrieve annotations.
+
+        :param idx: index of the dataset item for which to get the annotations
+        :return ann_info: dict that contains the coordinates of the bboxes and their corresponding labels
+        """
+
+        dataset_item = self.otx_dataset[idx]
+        ann_info = get_annotation_mmseg_format(dataset_item, self.project_labels)
+
+        return ann_info
+
+    @check_input_parameters_type()
+    def get_gt_seg_maps(self, efficient_test: bool = False):
+        """Get ground truth segmentation maps for evaluation."""
+
+        gt_seg_maps = []
+        for item_id in range(len(self)):
+            ann_info = self.get_ann_info(item_id)
+            gt_seg_maps.append(ann_info["gt_semantic_seg"])
+        if efficient_test:
+            pass
+
+        return gt_seg_maps
+
+
+@DATASETS.register_module()
+class MPASegDataset(OTXSegDataset):
```

Comment:
Is there specific reason for separating MPASegDataset with OTXSegDataset?

Reply:
some parameter conflict in segmentation, This seems to require alignment with the cleanup of the dataset configuration. 
I tried to merge the two classes, but for now I left it as is.